modules/core/org.eclipse.fx.osgi/src/main/java/org/eclipse/fx/osgi/fxloader/FXClassLoader.java

```diff
@@ -444,7 +444,8 @@ private static ModuleLayerWrapper advancedModuleLayerBoostrap(ClassLoader parent
 		@SuppressWarnings("deprecation")
 		URL[] urls = Stream.of(paths).map( Path::toFile).map( f -> {
 			try {
-				return f.toURL();
+				final URL url = f.toURL();
+				return new URL(url.toExternalForm().replace(" ", "%20")); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
I think the right way is to use `f.toURI().toURL()`

Reply:
You are right